### PR TITLE
Ability to modify OIDC provider configuration

### PR DESCRIPTION
**Pull request by ovidiupopa91**
Closes gh-616

### CODE REVIEW

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java [hunk 1]
```diff
@@ -103,6 +104,20 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
...
+	/**
+	 * Sets the {@link HttpMessageConverter} used for converting the {@link OidcProviderConfiguration}
+	 * from and to HTTP requests and responses
+	 *
+	 * @param providerConfigurationHttpMessageConverter the {@link HttpMessageConverter} used for converting a
+	 * representation of the OpenID Provider Configuration from and to HTTP requests and responses
+	 *
+	 * @since 0.2.3
+	 */
+	public void setProviderConfigurationHttpMessageConverter(HttpMessageConverter<OidcProviderConfiguration> providerConfigurationHttpMessageConverter) {
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:116]: The thing about exposing `HttpMessageConverter<OidcProviderConfiguration>` is that it handles both read and write, but this `Filter` will only ever use the write. I wonder if we should just expose `Converter<OidcProviderConfiguration, Map<String, Object>>`, which ultimately hooks into `OidcProviderConfigurationHttpMessageConverter.setProviderConfigurationParametersConverter()`

Comment (ovidiupopa91) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:116]: hi @jgrandja. I was thinking about this as well when I wrote the JavaDoc for the setter :-) 

I think it's a good idea to only expose the setter for the converter as the read method will never be used.

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java [hunk 2]
```diff
@@ -103,6 +105,21 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
...
+	/**
+	 * Sets the {@link Converter} used for converting the {@link OidcProviderConfiguration} to a
+	 * {@code Map} representation of the OpenID Provider Configuration.
+	 *
+	 * @param providerConfigurationParametersConverter the {@link Converter} used for converting to a
+	 * {@code Map} representation of the OpenID Provider Configuration
+	 *
+	 * @since 0.2.3
+	 */
+	public void setProviderConfigurationParametersConverter(
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:117]: Let's rename to `setProviderConfigurationResponseConverter()`

File: oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilterTests.java [hunk 3]
```diff
@@ -71,6 +84,53 @@ public void doFilterWhenNotConfigurationRequestThenNotProcessed() throws Excepti
 
 		verify(filterChain).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 	}
+	@Test
+	public void providerConfigurationParametersConverterWhenCustomThenAbleToOverride() throws Exception{
```
Comment (jgrandja) [inline oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilterTests.java:88]: Please rename method to be consistent with other methods...should start with `doFilter...`

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the updates @ovidiupopa91. Please see review comments.

As well, we should add a customization hook in `OidcConfigurer`, for example `providerConfigurationResponseConverter()`

**Comment by jgrandja** [comment]
Hi @ovidiupopa91. I'm just circling back to this PR now. Are you still available to work on this? No worries if you don't have time. Please let me know either way. Thanks.

**Comment by jgrandja** [comment]
Hi @ovidiupopa91. I'm assuming you've been busy so I went ahead and merged this enhancement via 0994a1e1e14263d106373e3397f8c5c44f9a7708.

**Pull request closed without merging**